pipeline/pcler.py

```diff
@@ -118,188 +83,303 @@ def pcler(args):
+                        )
+
+                plt.savefig(f"{plot_dir}/decoupled_pcls_nobeam_{type}_{map_pair}_{split_pair}.pdf", # noqa
+                            bbox_inches="tight")
+            meta.timer.stop(
+                "plots",
+                text_to_output="Plot decoupled C_ells for mock data",
+                verbose=args.verbose
+            )
+
     if args.tf_est:
+
+        filtering_tags = meta.get_filtering_tags()
+        filtering_tag_pairs = meta.get_independent_filtering_pairs()
         for id_sim in range(meta.tf_est_num_sims):
-            fields = {"filtered": {}, "unfiltered": {}}
-            for pure_type in ["pureE", "pureB"]:
-                map_file = meta.get_map_filename_transfer2(
-                    id_sim,
-                    "tf_est",
-                    pure_type=pure_type
-                )
-                map_file_filtered = map_file.replace(".fits", "_filtered.fits")
+            meta.timer.start("pcler_tf_est")
 
-                map = hp.read_map(map_file, field=[0, 1, 2])
-                map_filtered = hp.read_map(map_file_filtered, field=[0, 1, 2])
+            fields = {ftag: {
+                "filtered": {},
+                "unfiltered": {}
+                } for ftag in filtering_tags
+            }
 
-                # TO-DO: filter temperature only once !
-                field = {
-                    "spin0": nmt.NmtField(mask, map[:1]),
-                    "spin2": nmt.NmtField(mask, map[1:],
-                                          purify_b=meta.tf_est_pure_B)
-                }
-                field_filtered = {
-                    "spin0": nmt.NmtField(mask, map_filtered[:1]),
-                    "spin2": nmt.NmtField(mask, map_filtered[1:],
-                                          purify_b=meta.tf_est_pure_B)
-                }
+            for ftag in filtering_tags:
+                for pure_type in ["pureE", "pureB"]:
+                    map_file = meta.get_map_filename_transfer(
+                        id_sim,
+                        "tf_est",
+                        pure_type=pure_type,
+                        filter_tag=ftag
+                    )
+                    map_file_filtered = map_file.replace(".fits",
+                                                         "_filtered.fits")
 
-                fields["unfiltered"][pure_type] = field
-                fields["filtered"][pure_type] = field_filtered
+                    map = hp.read_map(map_file, field=[0, 1, 2])
+                    map_filtered = hp.read_map(map_file_filtered,
+                                               field=[0, 1, 2])
 
-            pcls_mat_filtered = ps_utils.get_pcls_mat_transfer(
-                fields["filtered"],
-                nmt_binning
-            )
-            pcls_mat_unfiltered = ps_utils.get_pcls_mat_transfer(
-                fields["unfiltered"],
-                nmt_binning
-            )
+                    # TO-DO: filter temperature only once !
```

Comment:
Is this still an open issue?

Reply:
We need to include pureT maps to account for filtering induced T-P leakage